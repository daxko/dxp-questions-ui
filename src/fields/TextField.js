var React = require('react');

var TextField = React.createClass({

	onKeyPress: function(e) {
		if (e.key == "<" || e.key == ">")
			e.preventDefault();
	},

	render: function() {

			return (
				<input type="text"
					maxLength={this.props.question.max_length}
					readOnly={this.props.question.read_only}
					value={this.props.answer || '' }
					onChange={this.props.onChange}
					onKeyPress={this.onKeyPress}
					className={this.props.className}
					onBlur={this.props.onBlur}
				/>
			);
	}
	

});

module.exports = TextField;
