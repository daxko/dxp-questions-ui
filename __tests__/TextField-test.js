import React from 'react'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import TextField from '../src/fields/TextField'
const json = require('../snippets/json/text_question.json')

configure({adapter: new Adapter() })

describe('TextField component', () => {
  it('should render a question', () => {
    const wrapper = shallow(<TextField question={json} possible_answers={json.possible_answers}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
