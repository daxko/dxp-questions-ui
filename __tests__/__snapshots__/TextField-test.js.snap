// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component should render a question 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField
    possible_answers={undefined}
    question={
      Object {
        "description": "For example, peanuts, dairy, eggs, gluten, wheat, etc.",
        "max_length": 50,
        "read_only": false,
        "required": false,
        "title": "Please specify any food allergies",
        "type": "text",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "maxLength": 50,
      "onBlur": undefined,
      "onChange": undefined,
      "onKeyPress": [Function],
      "readOnly": false,
      "type": "text",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "maxLength": 50,
        "onBlur": undefined,
        "onChange": undefined,
        "onKeyPress": [Function],
        "readOnly": false,
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
